pandas/core/arrays/period.py

```diff
@@ -757,7 +759,12 @@ def _add_timedeltalike_scalar(self, other):
         assert isinstance(self.freq, Tick)  # checked by calling function
         assert isinstance(other, (timedelta, np.timedelta64, Tick))
 
-        delta = self._check_timedeltalike_freq_compat(other)
+        if isna(other):
```

Comment:
I find the need to do this pretty odd, why would't the check routine handle this? I do not like special cases

Reply:
Because the first thing the check function does is call `delta_to_nanoseconds`, which will raise if we pass `np.timedelta64("NaT")`.  Even if we change that method to avoid calling `delta_to_nanoseconds` in that case, we'd need to handle a null return value after the check.

> I do not like special cases

+1.  As Joris mentioned in another thread, a lot of gymnastics seems to be driven by the fact that `pd.NaT` does double-duty as both Not-A-Time and Not-A-Timedelta.